src/gsi/read_satwnd.f90

```diff
@@ -923,10 +949,79 @@ subroutine read_satwnd(nread,ndata,nodata,infile,obstype,lunout,gstime,twind,sis
+                      else if(hdrdat(9) >= four) then             ! WV deep layer,monitoring
+                         itype=254
+                         qm=9                                     !  quality mark as 9, means the observation error needed to be set
+                         c_station_id='WV'//stationid
+                         c_sprvstg='WV'
+                      endif
+!  get quality information THIS SECTION NEEDS TO BE TESTED!!!
+                      call ufbint(lunin,rep_array,1,1,iret, '{AMVIVR}')
+                      irep_array = int(rep_array)
+                      allocate( amvivr(2,irep_array))
+                      call ufbrep(lunin,amvivr,2,irep_array,iret, 'TCOV CVWD')
+                      pct1 = amvivr(2,1)     ! use of pct1 (a new variable in the BUFR) is introduced by Nebuda/Genkova
+                      deallocate( amvivr )
+
+                      call ufbseq(lunin,amvqic,2,4,iret, 'AMVQIC') ! AMVQIC:: GNAPS PCCF
+                      qifn = amvqic(2,2)  ! QI w/ fcst does not exist in this BUFR
+                      ee = amvqic(2,4) ! NOTE: GOES-R's ee is in [m/s]
+                      if(qifn <85.0_r_kind )  then    !  qifn, QI without forecast
+                         qm=15
+                      endif
+              endif
+! Extra block for new EUMETSAT BUFR: End
+! Extra block for new Metop/AVHRR BUFR: Start
+           else if(trim(subset) == 'NC005081') then         ! Metop/AVHRR from NESDIS
+              if( hdrdat(1) <10.0_r_kind .or. (hdrdat(1) >= 200.0_r_kind .and. &
+                   hdrdat(1) <=223.0_r_kind)  ) then  ! The range of satellite IDs
+                 c_prvstg='AVHRR'
+                 if(hdrdat(9) == one)  then                            ! IRwinds
+                    itype=244
+                    c_station_id='IR'//stationid
+                    c_sprvstg='IR'
+                 else
+                    write(6,*) 'READ_SATWND: wrong derived method value'
+                 endif
+
+                 call ufbint(lunin,rep_array,1,1,iret, '{AMVIVR}')
+                 irep_array = int(rep_array)
+                 allocate( amvivr(2,irep_array))
+                 call ufbrep(lunin,amvivr,2,irep_array,iret, 'TCOV CVWD')
+                 pct1 = amvivr(2,1)     ! use of pct1 is limited to GOES-16/17) as introduced by Nebuda/Genkova
+                 deallocate( amvivr )
+
+                 call ufbseq(lunin,amvqic,2,4,iret, 'AMVQIC') ! AMVQIC:: GNAPS PCCF
+                 qifn = amvqic(2,2)  ! QI w/ fcst does not exist in this BUFR
+                 ee = amvqic(2,4) ! NOTE: GOES-R's ee is in [m/s]
+              endif
+! Extra block for new Metop/AVHRR BUFR: End
+
 ! Extra block for VIIRS NOAA-20: Start
            else if(trim(subset) == 'NC005091') then
-              if(hdrdat(1) >=r250 .and. hdrdat(1) <=r299 ) then  ! The range of satellite IDs
+              if( hdrdat(1) >=r200 .and. hdrdat(1) <=r250  ) then  ! The range of satellite IDs
```

Comment:
Please revert this change as per management decision.   I recommend leaving the correct code in read_satwnd.f90 - just comment it out.   The corrected code will be activated in a later implementation.